### PR TITLE
Dev.del/fix-editor-translations

### PR DESCRIPTION
**Pull request by deltork**
### PR Goal? 
Complete the implementation of the translation management in the editor

### Fixes? 
Various issues with the translations in the editor

### Feedback sought? 
Adding/remove/updating translations in the studio and editor. Also finding missing edge cases.

### Priority? 
HIGH

### Tests added? 
Will be part of the studio test suites

### How to test? 
1. Create a read-along, add some translations. 
2. Export the single file format
3. Upload it to the editor
4. Edit the translations 
5. Export the translations

### Confidence? 
HIGH

### Version change? 
patch

This patch does not use the agreed upon model we discussed. It just fixes the issues so we can proceed to adding tests before the work to change the modeling is done.

### CODE REVIEW

**Comment by deltork** [comment]
@joanise I have streamlined the translation management as discussed.

File: packages/web-component/src/components/read-along-component/read-along.tsx [hunk 1]
```diff
@@ -1074,6 +1079,40 @@ export class ReadAlongComponent {
...
+                  if (sentence.id && sentenceIsAligned(sentence)) {
+                    lastAlignedSentenceId = `${sentence.id}`;
+                  } else if (/translation/.test(sentence.className)) {
+                    const translation: { [key: string]: string } = {};
+                    if ((sentence as Element).hasAttribute("sentence-id")) {
+                      let sentenceID = (sentence as Element).getAttribute(
+                        "sentence-id",
+                      );
+
+                      translation[sentenceID] = sentence.textContent;
```
Comment (joanise) [inline packages/web-component/src/components/read-along-component/read-along.tsx:1099]: We maybe want to null out `lastAlignedSentenceId` here too, I think.

**Review by joanise — APPROVED** [review]
This works great, thank you! I tested adding, updating, removing translations, in the editor and studio, and the other edit features, and things seem to be working correctly. This brings the Studio&Editor in much better shape!
